### ISSUE TITLE
plz help me

### ISSUE THREAD

**Issue by shanuabbas-prog** [gensyn-ai/rl-swarm#221]
Generating train split: 4 examples [00:00, 1146.38 examples/s]
Map: 100%|██████████████████████████| 4/4 [00:00<00:00, 422.83 examples/s]
  0%|                                              | 0/20 [00:00<?, ?it/s]/home/user/rl-swarm/.venv/lib/python3.12/site-packages/torch/utils/data/dataloader.py:665: UserWarning: 'pin_memory' argument is set as true but no accelerator is found, then device pinned memory won't be used.
  warnings.warn(warn_msg)
`use_cache=True` is incompatible with gradient checkpointing. Setting `use_cache=False`.
/home/user/rl-swarm/.venv/lib/python3.12/site-packages/transformers/trainer.py:3700: FutureWarning: `torch.cpu.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cpu', args...)` instead.
  ctx_manager = torch.cpu.amp.autocast(cache_enabled=cache_enabled, dtype=self.amp_dtype)

**Comment by BenFielding** [comment]
This looks fine
